dojo/filters.py

```diff
@@ -2961,10 +2972,161 @@ def __init__(self, *args, **kwargs):
+        label="Product Type Name Contains",
+        help_text="Search for Product Type names that contain a given pattern")
+    test__engagement__product__name = CharFilter(
+        field_name="test__engagement__product__name",
+        lookup_expr="iexact",
+        label="Product Name",
+        help_text="Search for Product names that are an exact match")
+    test__engagement__product__name_contains = CharFilter(
+        field_name="test__engagement__product__name",
+        lookup_expr="icontains",
+        label="Product name Contains",
+        help_text="Search for Product Typ names that contain a given pattern")
+    test__engagement__name = CharFilter(
+        field_name="test__engagement__name",
+        lookup_expr="iexact",
+        label="Engagement Name",
+        help_text="Search for Engagement names that are an exact match")
+    test__engagement__name_contains = CharFilter(
+        field_name="test__engagement__name",
+        lookup_expr="icontains",
+        label="Engagement name Contains",
+        help_text="Search for Engagement names that contain a given pattern")
+    test__name = CharFilter(
+        field_name="test__name",
+        lookup_expr="iexact",
+        label="Test Name",
+        help_text="Search for Test names that are an exact match")
+    test__name_contains = CharFilter(
+        field_name="test__name",
+        lookup_expr="icontains",
+        label="Test name Contains",
+        help_text="Search for Test names that contain a given pattern")
+
+    def __init__(self, *args, **kwargs):
+        self.manage_kwargs(kwargs)
+        super().__init__(*args, **kwargs)
+
+        product_type_refs = [
+            "test__engagement__product__prod_type__name",
+            "test__engagement__product__prod_type__name_contains",
+        ]
+        product_refs = [
+            "test__engagement__product__name",
+            "test__engagement__product__name_contains",
+            "test__engagement__product__tags",
+            "test__engagement__product__tags_contains",
+            "not_test__engagement__product__tags",
+            "not_test__engagement__product__tags_contains",
+        ]
+        engagement_refs = [
+            "test__engagement__name",
+            "test__engagement__name_contains",
+            "test__engagement__tags",
+            "test__engagement__tags_contains",
+            "not_test__engagement__tags",
+            "not_test__engagement__tags_contains",
+        ]
+        test_refs = [
+            "test__name",
+            "test__name_contains",
+            "test__tags",
+            "test__tags_contains",
+            "not_test__tags",
+            "not_test__tags_contains",
+        ]
+
+        if self.test:
+            self.delete_tags_from_form(product_type_refs)
+            self.delete_tags_from_form(product_refs)
+            self.delete_tags_from_form(engagement_refs)
+            self.delete_tags_from_form(test_refs)
+        if self.engagement:
```

Comment:
Seems like this should be an `elif` since it won't have any effect if the condition above was true.

```suggestion
        elif self.engagement:
```